docs/DeveloperGuide.adoc

```diff
@@ -0,0 +1,665 @@
+
+image::atas help command sequence diagram v3.PNG[Component interactions for help command]
+
+1. The `Ui` class is used to read user input. +
+2. The `Parser` class is used to parse the user input string, returning a `Command` object. +
+3. The `Command#execute()` method is run, returning a `CommandResult` object. +
+4. The `Ui` class is used to show the `CommandResult` message to the user. +
+5. The `Storage` object is used to save the new state of the application.
+
+== Setting Up
+
+Ensure you have Java 11 or above installed on your computer +
+
+* For *Windows* Users:
+... Download the latest release of *ATAS* https://github.com/AY1920S2-CS2113T-M16-1/tp/releases[here].
+... Open a `cmd` (Command Prompt) window.
+... Navigate to the folder containing downloaded jar file.
+... Run the command `java -jar atas.jar`. You will be greeted with the welcome screen of *ATAS* in a few seconds.
+
+* For *Mac* Users:
+... Download the latest release of *ATAS* https://github.com/AY1920S2-CS2113T-M16-1/tp/releases[here].
+... Open up `Terminal`
+... Navigate to the directory containing downloaded jar file.
+... Run the command `java -jar atas.jar`. You will be greeted with the welcome screen of *ATAS* in a few seconds.
+
+== Implementation
+This section will detail how some noteworthy features are implemented.
+
+=== Delete Task Feature
+
+Current Implementation: +
+The `DeleteCommand` extends the `Command` class and initializes the `delete index` in its constructor. The `delete index`
+specifies the index of task that the user wants to delete.
+
+Given below is an example usage and how the `DeleteCommand` mechanism behaves at each step:
+
+*Step 1* +
+The user launches the app and retrieves the tasks which are saved under a local file using `Storage`.
+
+*Step 2* +
+The user enters `delete 2` into the command line. Method `parseCommand()` from the `Parser` class will be called to parse the command
+provided. It will obtain information to get `delete index`.
+
+[WARNING]
+If `IndexOutOfBoundsException` or `NumberFormatException` is caught, a new `IncorrectCommand` class will be called to
+print the respective error messages
+
+*Step 3* +
+A new instance of `DeleteCommand` with `delete index` initialized will be created. The `execute()` method of
+`DeleteCommand` will then be called.
+
+*Step 4* +
+The `execute()` method will do 2 things:
+
+* If there are no tasks in the existing task list, it will initialize a new `CommandResult` class that prints out an error
+message indicating an empty task list
+* If there are tasks in the existing task list, the `DeleteCommand` class will call the `deleteTask()` method from the
+`TaskList` class to delete the task, based on the index. At the end of the execution, the `DeleteCommand` class will
+initialize a new `CommandResult` class that prints out the success message for task deletion.
+
+The following sequence diagram summarizes how delete command operation works: +
+
+image::delete.png[delete task]
```

Comment:
1) Maybe the condition label on the opt block can be improved to say that the list is empty instead of just mentioning a "0" for easier readability.

![image](https://user-images.githubusercontent.com/59698024/77986576-0660c200-734a-11ea-89e3-f48b9f245c3c.png)